blocks/category-carousel-block/features/category-carousel/default.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { useComponentContext, useFusionContext } from "fusion:context";
+import getProperties from "fusion:properties";
+import getTranslatedPhrases from "fusion:intl";
+
+import PropTypes from "@arc-fusion/prop-types";
+import { Carousel, Heading, HeadingSection, Icon, Image } from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-category-carousel";
+const MIN_SLIDES = 4;
+const MAX_SLIDES = 12;
+
+const customFieldGroups = Array(MAX_SLIDES)
+	.fill(MIN_SLIDES)
+	.map((minimum, index) => ({
+		[`imageUrl_${index}`]: PropTypes.string.tag({
+			label: "Image",
+			description: "Select an image for this item.",
+			searchable: "image",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+		[`label_${index}`]: PropTypes.string.tag({
+			label: "Label",
+			description: "Enter a category label for this item.",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+		[`linkUrl_${index}`]: PropTypes.string.tag({
+			label: "Link Url",
+			description: "Enter a url to navigate to when selected.",
+			group: `Category ${index + 1}${index < minimum ? " (required)" : ""}`,
+			required: index < minimum,
+		}),
+	}));
+
+const mergeObjectArray = (accumulator, item) => ({ ...accumulator, ...item });
+
+function CategoryCarousel({ customFields }) {
+	const { headerText } = customFields;
+
+	const { id } = useComponentContext();
+	const { arcSite } = useFusionContext();
+	const { locale = "en" } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+
+	const validCategoryData = customFieldGroups
+		.map((fieldDefinition) =>
+			Object.keys(fieldDefinition)
+				.map((field) => ({ [field.replace(/_\d+$/, "")]: customFields[field] }))
+				.reduce(mergeObjectArray)
+		)
+		.filter((fields) => fields.imageUrl && fields.label && fields.linkUrl);
+
+	return validCategoryData.length >= MIN_SLIDES ? (
+		<HeadingSection>
```

Comment:
You only need this `<HeadingSection>` if the `headerText` has content

Reply:
Fixed in latest

Reply:
I put that in, but now that I did I remembered that the slide is not "valid" unless all the fields are present, so the header not being there will never be a concern.

Reply:
See #1320 as an example of what I was referring to with regards to only needing the `<HeadingSection>` if the block header is in use

Reply:
Implemented this change.  I was looking at the wrong header in the comment above.